Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM node:18-alpine as base
+RUN apk add --no-cache g++ make py3-pip libc6-compat
+WORKDIR /app
+COPY package*.json ./
+EXPOSE 3000
+
+FROM base as builder
+WORKDIR /app
+RUN npm install
+COPY . .
+RUN npm run build
```

Comment:
not sure why the base exists without having even the src file in there
i'd recommend taking a look at this https://medium.com/@maheshwar.ramkrushna/optimized-docker-multistage-build-for-node-js-application-deployment-with-nginx-17655b9e62c1
the idea is pretty similar
your base has the source code copied over with the build command. which builds the artifact
the second stage is just copyinig over the build artifact over to the production image which should be as minimalistic as possible with the least amount of permissions (similar to what you have below)